src/compiler/compiler.ts

```diff
@@ -1,40 +1,38 @@
-'use strict';
-
-var summary = module.exports = require('datalib/src/stats').summary;
-
-require('../globals');
-
 /**
  * Module for compiling Vega-lite spec into Vega spec.
  */
-var compiler = module.exports = {};
-
-var Encoding = require('../Encoding'),
-  axis = compiler.axis = require('./axis'),
-  legend = compiler.legend = require('./legend'),
-  marks = compiler.marks = require('./marks'),
-  scale = compiler.scale = require('./scale');
-
-compiler.data = require('./data');
-compiler.facet = require('./facet');
-compiler.layout = require('./layout');
-compiler.stack = require('./stack');
-compiler.style = require('./style');
-compiler.subfacet = require('./subfacet');
-compiler.time = require('./time');
-
-compiler.compile = function (spec, stats, theme) {
-  return compiler.compileEncoding(Encoding.fromSpec(spec, theme), stats);
+
+import {summary} from '../util';
+import {Encoding} from '../Encoding';
+
+import * as axis from './axis';
```

Comment:
The thing is that we often use local variables and call functions with the same name. We should rename the functions/modules as you suggest.